e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -313,7 +310,7 @@ await deviceClient
                     .InvokeAsync(deviceId, directMethodRequest)
                     .ConfigureAwait(false);
             }
-            catch (DeviceNotFoundException)
```

Comment:
This was incorrect as the root cause here was unsubscribing the direct method from the device, so [DeviceNotOnline ](https://docs.microsoft.com/en-us/azure/iot-hub/troubleshoot-error-codes#404103-devicenotonline)should be expected instead.

Reply:
`DeviceNotOnline` was not existing before, so adding within this PR.